src/lib.rs

```diff
@@ -1049,6 +1051,34 @@ pub enum FileMode {
     Commit,
 }
 
+impl From<u32> for FileMode {
+    fn from(mode: u32) -> Self {
+        match mode {
+            raw::GIT_FILEMODE_UNREADABLE => FileMode::Unreadable,
+            raw::GIT_FILEMODE_TREE => FileMode::Tree,
+            raw::GIT_FILEMODE_BLOB => FileMode::Blob,
+            raw::GIT_FILEMODE_BLOB_EXECUTABLE => FileMode::BlobExecutable,
+            raw::GIT_FILEMODE_LINK => FileMode::Link,
+            raw::GIT_FILEMODE_COMMIT => FileMode::Commit,
+            mode => panic!("unknown file mode: {}", mode),
+        }
+    }
+}
+
+impl Into<u32> for FileMode {
```

Comment:
FWIW this is typically phrase as `From<FileMode> for u32`, but if `FileMode` is an `enum` I think it would be better to set the discriminants to the `raw::*` values and that way `enum_value as u32` should work for this purpose.

Reply:
So we don't need `Into<u32>` trait implementation but just use `file_mode as u32` when needed?